internal/app/singularity/plugin_compile_linux.go

```diff
@@ -7,18 +7,27 @@ package singularity
 
 import (
 	"fmt"
+	"go/build"
 	"os"
 	"os/exec"
 	"path/filepath"
 	"runtime"
+	"strings"
 
 	uuid "github.com/satori/go.uuid"
 	"github.com/sylabs/sif/pkg/sif"
 )
 
+var (
+	workpath   = filepath.Join(build.Default.GOPATH, repo)
+	trimpath   = filepath.Dir(workpath)
```

Comment:
>Isn't this too much? If build.Default.GOPATH = "/home/user/go", workpath = "/home/user/go/src/github.com/sylabs/singularity" and trimpath is "/home/user/go/src/github.com/sylabs".
>I would have expected that trimming "/home/user/go" or "/home/user/go/src" would have been enough. What's the supported use case?

Yes, I think that trimming `$(GOPATH)/src` is actually probably enough. However, currently in the upstream code we are trimming `dirname $(SOURCEDIR)` from the singularity builds. I would vote we merge this as is to get something up and running, and revisit that in a follow-up PR.

> Also note that GOPATH is a PATH, meaning it can contain multiple directories separated by colons.

Huh, I guess I technically knew that but figured that `build.Default.GOPATH` would properly return one path rather than a list, but it looks like that's not the case.

Reply:
Ok, that makes sense.